vllm/distributed/kv_transfer/kv_connector/v1/nixl_connector.py

```diff
@@ -340,6 +369,7 @@ def register_kv_caches(self, kv_caches: dict[str, torch.Tensor]):
             # MLA case.
             self.num_blocks = first_kv_cache.shape[0]
             block_rank = 2  # [block_size, latent_dim]
+            # TODO does this include tp dependent size?
```

Comment:
For MLA we replicate the KV cache across TP ranks, so in this case the prefiller would need to send the same blocks to all decoders.